daml-lf/api-type-signature/src/test/scala/lf/typesig/reader/SignatureReaderSpec.scala

```diff
@@ -325,8 +341,26 @@ class SignatureReaderSpec extends AnyWordSpec with Matchers with Inside {
     }
 
     "resolve retro implements harmlessly when there are none" in {
+      // TODO: CL This one shouldn't pass imo
       PackageSignature.resolveRetroImplements((), itp.all)((_, _) => None) should ===((), itp.all)
```

Comment:
I think this line is not expected. it should not be the same. cc @S11001001

Reply:
You need to do what you did for `itpES`.

Reply:
Actually, I take it back. It absolutely should be the same, and the test is fine like it is.

Reply:
To explain, the caller is in control of resolution. By passing in `(_, _) => None` they're making it so that resolution always happens, but always fails. The test ensures that even when resolution fails, we don't get confused about the data structure, we leave the retroImplements there in case we're incrementally loading.

Reply:
@chunlokling-da demonstrated how I was misreading the result semantics of `resolveRetroImplements` and how the implementation mismatched the intent as a result; we fixed it in pairing.